ocs_ci/utility/vsphere.py

```diff
@@ -290,6 +294,51 @@ def add_disks(self, num_disks, vm, size, disk_type='thin'):
         for _ in range(int(num_disks)):
             self.add_disk(vm, size, disk_type)
 
+    def add_rdm_disk(self, vm, device_name, disk_mode=None, compatibility_mode=None):
+        """
+        Attaches RDM disk to vm
+
+        Args:
+            vm (vim.VirtualMachine): VM instance
+            device_name (str): Device name to add to VM.
+                e.g:"/vmfs/devices/disks/naa.600304801b540c0125ef160f3048faba"
+            disk_mode (str): Disk mode. By default it will
+                add 'independent_persistent'. Available modes are 'append',
+                'independent_nonpersistent', 'independent_persistent',
+                'nonpersistent', 'persistent', and 'undoable'
+            compatibility_mode (str): Compatabilty mode. Either 'physicalMode'
+                or 'virtualMode'. By default it will add 'physicalMode'.
+
+        """
+        logger.info(f"Adding RDM disk {device_name} to {vm.config.name}")
+        if not disk_mode:
+            disk_mode = DISK_MODE
+        if not compatibility_mode:
+            compatibility_mode = COMPATABILITY_MODE
+
+        spec = vim.vm.ConfigSpec()
+        controller = self.get_controller_for_adding_disk(vm)
+        unit_number = self.get_unit_number(vm)
+        logger.info(f"Unit number for new disk: {unit_number}")
+
+        device_changes = []
+        disk_spec = vim.vm.device.VirtualDeviceSpec()
+        disk_spec.fileOperation = "create"
+        disk_spec.operation = vim.vm.device.VirtualDeviceSpec.Operation.add
+        disk_spec.device = vim.vm.device.VirtualDisk()
+        rdm_info = vim.vm.device.VirtualDisk.RawDiskMappingVer1BackingInfo()
+        disk_spec.device.backing = rdm_info
+        disk_spec.device.backing.compatibilityMode = compatibility_mode
+        disk_spec.device.backing.diskMode = disk_mode
+        disk_spec.device.backing.deviceName = device_name
+        disk_spec.device.unitNumber = unit_number
+        disk_spec.device.controllerKey = controller.key
```

Comment:
Just thinking to add some more details to log in case we will need to debug it in future, it's probably worth to have more info in debug level maybe? But if you think it's not worth I am OK.

Reply:
I will include more in next PR